src/event/Event.php

```diff
@@ -47,6 +47,21 @@ final public function getEventName() : string{
 	 * @throws \RuntimeException if event call recursion reaches the max depth limit
 	 */
 	public function call() : void{
+		$this->callDepth(function(){
+			$handlers = HandlerListManager::global()->getHandlersFor(static::class);
+
+			foreach($handlers as $registration){
+				$registration->callEvent($this);
+			}
+		});
+	}
```

Comment:
This code is very performance sensitive. I don't like introducing extra function calls and closures into here.